AzureStorageRepositoryClientYamlTestSuiteIT test fails with SnapshotCreationException

Link to the build:

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.8+multijob+third-party-tests-azure-sas/5/console
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.8+multijob+third-party-tests-azure-sas/8/console
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.8+multijob+third-party-tests-azure-sas/9/console
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.8+multijob+third-party-tests-azure-sas/10/console

Command to reproduce:
```
./gradlew :plugins:repository-azure:qa:microsoft-azure-storage:integTestRunner \
  -Dtests.seed=8A38B5339259F4F5 \
  -Dtests.class=org.elasticsearch.repositories.azure.AzureStorageRepositoryClientYamlTestSuiteIT \
  -Dtests.method="test {yaml=repository_azure/10_repository/Snapshot/Restore with repository-azure}" \
  -Dtests.security.manager=true \
  -Dtests.locale=vi \
  -Dtests.timezone=Canada/Eastern \
  -Dcompiler.java=12 \
  -Druntime.java=8
```

Relevant excerpt from the logs:
```
12:42:46   1> [2019-07-22T12:42:45,372][INFO ][o.e.r.a.AzureStorageRepositoryClientYamlTestSuiteIT] [test] Stash dump on test failure [{
12:42:46   1>   "stash" : {
12:42:46   1>     "body" : {
12:42:46   1>       "error" : {
12:42:46   1>         "root_cause" : [
12:42:46   1>           {
12:42:46   1>             "type" : "snapshot_creation_exception",
12:42:46   1>             "reason" : "[repository:snapshot-one/bdo98uAzR9myTl-DjwVsdw] failed to create snapshot",
12:42:46   1>             "stack_trace" : "SnapshotCreationException[[repository:snapshot-one/bdo98uAzR9myTl-DjwVsdw] failed to create snapshot]; nested: StorageException[This request is not authorized to perform this operation.];
12:42:46   1> 	at org.elasticsearch.repositories.azure.AzureRepository.initializeSnapshot(AzureRepository.java:169)
12:42:46   1> 	at org.elasticsearch.snapshots.SnapshotsService$2.doRun(SnapshotsService.java:413)
12:42:46   1> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:751)
12:42:46   1> 	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
12:42:46   1> 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
12:42:46   1> 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
12:42:46   1> 	at java.base/java.lang.Thread.run(Thread.java:835)
12:42:46   1> Caused by: com.microsoft.azure.storage.StorageException: This request is not authorized to perform this operation.
12:42:46   1> 	at com.microsoft.azure.storage.StorageException.translateException(StorageException.java:87)
12:42:46   1> 	at com.microsoft.azure.storage.core.StorageRequest.materializeException(StorageRequest.java:313)
12:42:46   1> 	at com.microsoft.azure.storage.core.ExecutionEngine.executeWithRetry(ExecutionEngine.java:185)
12:42:46   1> 	at com.microsoft.azure.storage.blob.CloudBlobContainer.exists(CloudBlobContainer.java:744)
12:42:46   1> 	at com.microsoft.azure.storage.blob.CloudBlobContainer.exists(CloudBlobContainer.java:731)
12:42:46   1> 	at org.elasticsearch.repositories.azure.AzureStorageService.lambda$doesContainerExist$1(AzureStorageService.java:143)
12:42:46   1> 	at java.base/java.security.AccessController.doPrivileged(AccessController.java:551)
12:42:46   1> 	at org.elasticsearch.repositories.azure.SocketAccess.doPrivilegedException(SocketAccess.java:54)
12:42:46   1> 	at org.elasticsearch.repositories.azure.AzureStorageService.doesContainerExist(AzureStorageService.java:143)
12:42:46   1> 	at org.elasticsearch.repositories.azure.AzureBlobStore.containerExist(AzureBlobStore.java:98)
12:42:46   1> 	at org.elasticsearch.repositories.azure.AzureRepository.initializeSnapshot(AzureRepository.java:164)
12:42:46   1> 	... 6 more
```

--- a/plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java
+++ b/plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java
@@ -94,10 +94,6 @@ public class AzureBlobStore extends AbstractComponent implements BlobStore {
     public void close() {
     }
 
-    public boolean containerExist() throws URISyntaxException, StorageException {
-        return service.doesContainerExist(clientName, container);
-    }
-
     public boolean blobExists(String blob) throws URISyntaxException, StorageException {
         return service.blobExists(clientName, container, blob);
     }

--- a/plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureRepository.java
+++ b/plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureRepository.java
@@ -25,7 +25,6 @@ import com.microsoft.azure.storage.StorageException;
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-import org.elasticsearch.cluster.metadata.MetaData;
 import org.elasticsearch.cluster.metadata.RepositoryMetaData;
 import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.blobstore.BlobPath;
@@ -35,13 +34,9 @@ import org.elasticsearch.common.settings.Setting.Property;
 import org.elasticsearch.common.unit.ByteSizeValue;
 import org.elasticsearch.common.xcontent.NamedXContentRegistry;
 import org.elasticsearch.env.Environment;
-import org.elasticsearch.repositories.IndexId;
 import org.elasticsearch.repositories.blobstore.BlobStoreRepository;
-import org.elasticsearch.snapshots.SnapshotCreationException;
-import org.elasticsearch.snapshots.SnapshotId;
 
 import java.net.URISyntaxException;
-import java.util.List;
 import java.util.Locale;
 import java.util.function.Function;
 
@@ -155,20 +150,6 @@ public class AzureRepository extends BlobStoreRepository {
     @Override
     protected ByteSizeValue chunkSize() {
         return chunkSize;
-    }
-
-    @Override
-    public void initializeSnapshot(SnapshotId snapshotId, List<IndexId> indices, MetaData clusterMetadata) {
-        try {
-            final AzureBlobStore blobStore = (AzureBlobStore) blobStore();
-            if (blobStore.containerExist() == false) {
-                throw new IllegalArgumentException("The bucket [" + blobStore + "] does not exist. Please create it before "
-                        + " creating an azure snapshot repository backed by it.");
-            }
-        } catch (URISyntaxException | StorageException e) {
-            throw new SnapshotCreationException(metadata.name(), snapshotId, e);
-        }
-        super.initializeSnapshot(snapshotId, indices, clusterMetadata);
     }
 
     @Override

--- a/plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageService.java
+++ b/plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageService.java
@@ -137,12 +137,6 @@ public class AzureStorageService extends AbstractComponent {
         return prevSettings;
     }
 
-    public boolean doesContainerExist(String account, String container) throws URISyntaxException, StorageException {
-        final Tuple<CloudBlobClient, Supplier<OperationContext>> client = client(account);
-        final CloudBlobContainer blobContainer = client.v1().getContainerReference(container);
-        return SocketAccess.doPrivilegedException(() -> blobContainer.exists(null, null, client.v2().get()));
-    }
-
     public void deleteFiles(String account, String container, String path) throws URISyntaxException, StorageException {
         final Tuple<CloudBlobClient, Supplier<OperationContext>> client = client(account);
         // container name must be lower case.

--- a/plugins/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureStorageServiceMock.java
+++ b/plugins/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureStorageServiceMock.java
@@ -57,11 +57,6 @@ public class AzureStorageServiceMock extends AzureStorageService {
     }
 
     @Override
-    public boolean doesContainerExist(String account, String container) {
-        return true;
-    }
-
-    @Override
     public void deleteFiles(String account, String container, String path) throws URISyntaxException, StorageException {
         final Map<String, BlobMetaData> blobs = listBlobsByPrefix(account, container, path, null);
         for (String key : blobs.keySet()) {

